dev-proxy/ConsoleLogger.cs

```diff
@@ -336,58 +336,31 @@ public void Log<TState>(MSLogging.LogLevel logLevel, MSLogging.EventId eventId,
         var message = formatter(state, exception).ReplaceLineEndings();
         switch (logLevel)
         {
-            case MSLogging.LogLevel.Debug:
+            case LogLevel.Debug:
                 WriteDebug(message);
                 break;
-            case MSLogging.LogLevel.Information:
+            case LogLevel.Information:
                 WriteLog(message);
                 break;
-            case MSLogging.LogLevel.Warning:
+            case LogLevel.Warning:
                 WriteWarning(message);
                 break;
-            case MSLogging.LogLevel.Error:
+            case LogLevel.Error:
                 WriteError(message);
                 break;
         }
         
     }
 
     /// <inheritdoc/>
-    public bool IsEnabled(MSLogging.LogLevel logLevel) => CurrentLogLevel switch
-    {
-        // Current log level is Debug, so all log levels are enabled
-        MSLogging.LogLevel.Debug => true,
-        // Current log level is Info, so only Info, Warning, and Error log levels are enabled
-        MSLogging.LogLevel.Information => logLevel >= MSLogging.LogLevel.Information,
-        // Current log level is Warn, so only Warning and Error log levels are enabled
-        MSLogging.LogLevel.Warning => logLevel >= MSLogging.LogLevel.Warning,
-        // Current log level is Error, so only Error log level is enabled
-        MSLogging.LogLevel.Error => logLevel >= MSLogging.LogLevel.Error,
-        // Current log level is not recognized, so no log levels are enabled
-        _ => false
-    };
+    public bool IsEnabled(LogLevel logLevel) => logLevel >= _currentLogLevel; // only log if the log level is greater than or equal to the current log level
 
     /// <inheritdoc/>
     public IDisposable? BeginScope<TState>(TState state) where TState : notnull => default!;
 
-    public void SetLogLevel(LogLevel logLevel)
+    public LogLevel LogLevel
     {
-        CurrentLogLevel = logLevel switch
-        {
-            LogLevel.Debug => MSLogging.LogLevel.Debug,
-            LogLevel.Info => MSLogging.LogLevel.Information,
-            LogLevel.Warn => MSLogging.LogLevel.Warning,
-            LogLevel.Error => MSLogging.LogLevel.Error,
-            _ => throw new ArgumentOutOfRangeException(nameof(logLevel), logLevel, null)
-        };
+        get => _currentLogLevel;
```

Comment:
Is there a reason we need to define a private member instead of `public LogLevel LogLevel { get; set; }`?